utility/remoteaddr.go

```diff
@@ -91,3 +98,13 @@ func parseXForwardedFor(header http.Header) string {
 	}
 	return ""
 }
+
+func parseXOriginalForwardedFor(header http.Header) string {
```

Comment:
To be performant code instead of passing by value, we need to pass by reference.  All base code functions to be modified.

Reply:
Deferring this issue for now since all IP extraction functions currently use pass-by-value, we are not sure if pass-by-reference will cause any breakage